src/misc/support.cpp

```diff
@@ -399,6 +399,34 @@ const std_fs::path &GetExecutablePath()
 	return exe_path;
 }
 
+std_fs::path GetResource(const std_fs::path &name)
+{
+#if defined(MACOSX)
+	const auto bundled_path = GetExecutablePath() / "../Resources";
+#else
+	const auto bundled_path = GetExecutablePath() / "resources";
+#endif
+	const auto parents = {bundled_path,
+	                      // macOS, POSIX, and even MinGW/MSYS2/Cygwin:
+	                      std_fs::path("/usr/local/share/dosbox-staging"),
+	                      std_fs::path("/usr/share/dosbox-staging"),
+	                      std_fs::path(CROSS_GetPlatformConfigDir())};
```

Comment:
What's the best practice lookup order?

I would have thought home directories would be preferred over system directories for example.

Or am I reading this back-to-front?

Reply:
Prior to us bundling the translations, the user had to take care of all of DOSBox's add-on files - including downloading newer translations. They put everything in their config tree. 

But since we've started bundling the translations, we've offloaded that management burden (atleast for these files).

The "best" version of resource files are the ones most closely paired with the release package (and we really don't want users thinking they have to manage resource files.. imagine going to clang's data tree and mucking with their share objects and headers). 

Ideally I would prefer we only search our bundled resources dir and nothing else.. but unfortunately we have repo maintainers who will insist on moving data files in these /usr/share areas and not paired with the executable. 

So the order I have is in order of "newness".  First we look along side the binary itself (for users using the zipped release). Next is usr/local, which always trumps /usr, and at the end we check the user's config path - which is kind of a deprecated bucket. 

(Although for sf2, shaders, etc.. those aren't bundled resources - and that code still looks up primarily from the user's conf dir first).

Reply:
Fair enough.

If only C++ had a standardized resource embedding system, then resources could be "properly" embedded in the binary.

Reply:
Actually, maybe for bundled resources, the idea is that users shouldn't be managing them at all - so I think it makes sense to drop the user config directory entirely.

Reply:
> If only C++ had a standardized resource embedding system, then resources could be "properly" embedded in the binary.

Yeah; I suppose there are lots of libraries that bundle assets in big zip-line buckets instead of spraying thousands of tiny files into the user's OS (although I have seen some games deploy over 100,000 files! Craziness). 

But for other languages - I guess developers are given standardized way of packaging thousands of files into the binary itself? (Executables are tens of GB in size?) I'm not playing latest gen AAA games on Windows, so don't have much exposure.

Reply:
There are ways to do it in C++ as well. There's the Windows resource compiler system, Qt & WX made their own cross platform versions, plus various libraries that will embed resources as strings or hex blobs etc.

Other languages also have options, like Go's relatively new [embed](https://pkg.go.dev/embed) system.

I think games often use some sort of archive files for resources, they aren't usually embedded into the main binary though. Maybe they'll use something like [PhysicsFS](https://icculus.org/physfs/) for abstraction as well.

Reply:
Great info; thanks @shermp.